api/src/chat/services/conversation.service.ts

```diff
@@ -79,13 +84,26 @@ export class ConversationService extends BaseService<
     captureVars: boolean = false,
   ) {
     const msgType = event.getMessageType();
+    const profile = event.getSender();
     // Capture channel specific context data
     convo.context.channel = event.getHandler().getChannel();
     convo.context.text = event.getText();
     convo.context.payload = event.getPayload();
     convo.context.nlp = event.getNLP();
     convo.context.vars = convo.context.vars || {};
 
+    const contextVars = (
+      await this.contextVarService.find({
+        name: { $in: next.capture_vars.map((c) => c.context_var) },
+      })
+    ).reduce(
+      (acc, cv) => {
+        acc[cv.name] = cv;
+        return acc;
+      },
+      {} as Record<string, ContextVar>,
+    );
```

Comment:
I suggest we add this method to the "contextVarService" : getContextVarsByBlock(next)